cluster/src/dunit/scala/io/snappydata/externalstore/ExecutorMessageDUnitTest.scala

```diff
@@ -0,0 +1,165 @@
+  // server process in this test. Enable it after resolving the issue.
+  // When run individually, this test passes.
+  def _testStore02BlockMapUpdatesWithNodeDown(): Unit = {
+    executeSomething()
+    verifyMap("stopProcess")
+    // stopRest()
+    getLogWriter.info("testStore02BlockMapUpdatesWithNodeDown() Successful")
+  }
+
+  def executeSomething(): Unit = {
+    val snc = SnappyContext(sc)
+
+    val wc: WaitCriterion = new WaitCriterion {
+      override def done(): Boolean = {
+        SnappyContext.storeToBlockMap.size == 4 // 3 servers + 1 lead/driver
+      }
+      override def description(): String = {
+        s"Received executor message only from" +
+            s" ${SnappyContext.storeToBlockMap.size} members, expected from three."
+      }
+    }
+    DistributedTestBase.waitForCriterion(wc, 10000, 500, true)
+    assert(SnappyContext.storeToBlockMap.size == 4) // 3 servers + 1 lead/driver
+    for ((dm, blockId) <- SnappyContext.storeToBlockMap) {
+      assert(blockId != null)
+    }
+  }
+
+  def verifyMap(m: String): Unit = {
+    vm0.invoke(getClass, m)
+    assert(SnappyContext.storeToBlockMap.size == 3)
+    for ((dm, blockId) <- SnappyContext.storeToBlockMap) {
+      assert(blockId != null)
+    }
+
+    vm1.invoke(getClass, m)
+    assert(SnappyContext.storeToBlockMap.size == 2)
+    for ((dm, blockId) <- SnappyContext.storeToBlockMap) {
+      assert(blockId != null)
+    }
+
+    vm2.invoke(getClass, m)
+    assert(SnappyContext.storeToBlockMap.size == 1)
+  }
+
+  def stopRest(): Unit = {
+    import ExecutorMessageDUnitTest._
+    // vm3.invoke(getClass, "")
+    stopLocatorLead()
+  }
+}
+
+object ExecutorMessageDUnitTest {
+
+  def stopExecutor(): Unit = {
+    ExecutorInitiator.stop()
+    Thread.sleep(1000)
+  }
+
+  def restartExecutors(locatorPort: Int, props: Properties): Unit = {
+    SnappyContext.globalSparkContext.stop()
+    ClusterManagerTestBase.startSnappyLead(locatorPort, props)
+  }
+
+  def stopProcess(): Unit = {
+    // ClusterManagerTestBase.stopAny()
+    ClusterManagerTestBase.cleanupTestData(null, null)
+    // ClusterManagerTestBase.stopNetworkServers()
+    ServiceManager.getServerInstance.stop(null)
+    // Thread.sleep(2000)
+  }
+
+  def stopLocatorLead(): Unit = {
+    // Stop lead
+    SnappyContext.globalSparkContext.stop()
+    val lead = ServiceManager.getServerInstance
+    if (lead.status() == FabricService.State.RUNNING) {
+      lead.stop(null)
+    }
+    // Stop locator
+    DistributedTestBase.invokeInLocator(new SerializableRunnable() {
+      override def run(): Unit = {
+        val loc = ServiceManager.getLocatorInstance
+        if (loc.status() == FabricService.State.RUNNING) {
+          loc.stop(null)
+        }
+      }
+    })
+    // Delete all oplog files
+    deleteOplogFiles(true)
+  }
+
+  @throws(classOf[IOException])
+  def deleteOplogFiles(includeDataDictionary: Boolean): Unit = {
```

Comment:
Also, we should be very wary about start and stop issues in snappy as that has been the pain point for gemxd customers.